azurerm/internal/services/datafactory/data_factory_linked_service_azure_databricks_resource.go

```diff
@@ -0,0 +1,598 @@
+						},
+					},
+				},
+				ExactlyOneOf: []string{"existing_cluster_id", "new_cluster_config", "instance_pool"},
+			},
+
+			"integration_runtime_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"annotations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"additional_properties": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceDataFactoryLinkedServiceDatabricksCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.LinkedServiceClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	dataFactoryName := d.Get("data_factory_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Data Factory Linked Service Databricks %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory_linked_service_data_lake_storage_gen2", *existing.ID)
+		}
+	}
+
+	var DatabricksProperties *datafactory.AzureDatabricksLinkedServiceTypeProperties
+
+	// Check if the MSI authentication block is set
+	msiAuth := d.Get("authentication_msi").([]interface{})
+	accessTokenAuth := d.Get("authentication_access_token").([]interface{})
+	accessTokenKeyVaultAuth := d.Get("authentication_key_vault_password").([]interface{})
+
+	if len(msiAuth) > 0 {
+		// MSI Auth map has data
+		workspaceResourceID := msiAuth[0].(map[string]interface{})["workspace_resource_id"].(string)
+
+		DatabricksProperties = &datafactory.AzureDatabricksLinkedServiceTypeProperties{
+			Authentication:      "MSI",
+			WorkspaceResourceID: workspaceResourceID,
+		}
+	}
+	if len(accessTokenAuth) > 0 {
+		// Setup authentication using access tokens
+
+		accessToken := accessTokenAuth[0].(map[string]interface{})["access_token"].(string)
+
+		accessTokenAsSecureString := datafactory.SecureString{
+			Value: utils.String(accessToken),
+			Type:  datafactory.TypeSecureString,
+		}
+
+		// Assign the access token in the properties block
+		DatabricksProperties = &datafactory.AzureDatabricksLinkedServiceTypeProperties{
+			AccessToken: &accessTokenAsSecureString,
+		}
+	} else {
+		DatabricksProperties = &datafactory.AzureDatabricksLinkedServiceTypeProperties{
```

Comment:
I agree if's should be part of the same conditional (if/else if/else) rather than separate ifs. However in terms of initializing the properties to use the accessTokenKeyVaultAuth mechanism as a default, I don't think that will work because we don't know if they've provided the required key vault details. 

The only reason I'm not checking the length of the accessTokenKeyVaultAuth before using it in like 318is because in the schema I marked that we need exactly one of authentication_msi, authentication_access_token or authentication_key_vault_password, so by the time we get to the _else_ on line 317 above, we know it has to be set or the plan would have failed.

I'm thinking the fix here would be make them all part of the conditional (if/else if/else). Thoughts?

Reply:
The top part makes sense to me! But instead of doing `if/else` all the way down. We can just do `if` statements and scratch the else because we know one of them will work. It'll make it a bit easier to read too.

Reply:
Ah gotcha, I put them all as the same conditional since they are mutually exclusive, but since that is being checked by the ExactlyOneOf annotation, you're right, we should never get more than one of those set. Will adjust.